### PR TITLE
Add ServiceURL option for KMS client

### PR DESCRIPTION
**Pull request by bdurrani**
AmazonKeyManagementServiceConfig doesn't use the service url when it's setting on AmazonS3CryptoConfigurationV2

## Motivation and Context
I'm using LocalStack when I set ServiceURL on AmazonS3CryptoConfigurationV2 I'm seeing the Kms calling aws endpoint

See #29 for reference

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

**Comment by bdurrani** [comment]
@normj Hi there
Is this change ok for the KMS client `ServiceURL` option?
As discussed here: https://github.com/aws/amazon-s3-encryption-client-dotnet/pull/29

Thanks

File: src/AmazonS3CryptoConfigurationBase.cs [hunk 1]
```diff
@@ -41,5 +42,8 @@ public AmazonS3CryptoConfigurationBase()
             // By default, store encryption info in metadata
             StorageMode = CryptoStorageMode.ObjectMetadata;
         }
+
+        public AmazonKeyManagementServiceConfig KmsConfig { get; set; }
```
Comment (ashovlin) [inline src/AmazonS3CryptoConfigurationBase.cs:46]: Could you add XML documentation to this, maybe something like:
```
/// <summary>
/// Configuration for the internal AWS Key Management Service client that will be used internally when encrypting S3 objects with KMS keys.
/// </summary>
/// <remarks>
/// If not specified here, the internal KMS client will inherit the region, timeout, and proxy configuration from the S3 configuration
/// </remarks>
```

**Review by ashovlin — CHANGES_REQUESTED** [review]

File: src/AmazonS3EncryptionClientBase.cs [hunk 2]
```diff
@@ -60,6 +60,13 @@ internal IAmazonKeyManagementService KMSClient
                             {
                                 kmsConfig.SetWebProxy(proxySettings);
                             }
+
+                            var serviceConfig = this.S3CryptoConfig.KmsConfig;
+                            if ( serviceConfig != null
+                                 && !string.IsNullOrWhiteSpace(serviceConfig.ServiceURL))
+                            {
+                                kmsConfig.ServiceURL = serviceConfig.ServiceURL;
+                            }
```
[the remark this answers is not in the document]
Comment (bdurrani) [reply src/AmazonS3EncryptionClientBase.cs]: I think this is what you were looking for? Let me know if misunderstood.

Thanks

[the remark this answers is not in the document]
Comment (bdurrani) [reply src/AmazonS3EncryptionClientBase.cs]: I think this is what you were looking for? Let me know if misunderstood.

Thanks @ashovlin

**Review by ashovlin — APPROVED** [review]
Looks good to me, I'll line up a second reviewer from the internal team. We'll also handle the version bump once approved.

**Review by dscpinheiro — APPROVED** [review]

**Comment by ashovlin** [comment]
I've merged this to `dev`, which is our staging branch for the next release. We'll reply again once the new release is out.

**Comment by ashovlin** [comment]
This PR was released yesterday in [2.1.0](https://www.nuget.org/packages/Amazon.Extensions.S3.Encryption/2.1.0), thanks again for your contribution!

**Pull request merged**